app/domain/authentication/authn_jwt/signing_key/create_jwks_from_http_response.rb

```diff
@@ -0,0 +1,52 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # CreateJwksFromHttpResponse command class is responsible to create jwks object from http response
+      CreateJwksFromHttpResponse ||= CommandClass.new(
+        dependencies: {
+          logger: Rails.logger
+        },
+        inputs: [:http_response]
+      ) do
+        def call
+          validate_response_exists
+          create_jwks_from_http_respnse
+        end
+
+        private
+
+        def validate_response_exists
+          raise Errors::Authentication::AuthnJwt::MissingHttpResponse if @http_response.blank?
+        end
+
+        def create_jwks_from_http_respnse
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatingJwksFromHttpResponse.new)
+
+          raise Errors::Authentication::AuthnJwt::InvalidHttpResponseFormat unless @http_response.respond_to?(:body)
+
+          response_body = @http_response.body
+          encoded_body = Base64.encode64(response_body)
+          jwks, keys = parse_jwks_response(response_body, encoded_body)
+
+          if keys.blank?
+            raise Errors::Authentication::AuthnJwt::FetchJwksUriKeysNotFound, encoded_body
+          end
+
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatedJwks.new)
+          jwks
+        end
+
+        def parse_jwks_response(response_body, encoded_body)
+          parsed_response = JSON.parse(response_body)
+          keys = parsed_response['keys']
+          [{ keys: JSON::JWK::Set.new(keys) }, keys]
```

Comment:
`JSON::JWK::Set` should be a dependency

Reply:
instead of returning both `{ keys: JSON::JWK::Set.new(keys) }` and `keys` and then checking `if keys.blank?`, you can perform the check in this method and return only `{ keys: JSON::JWK::Set.new(keys) }`

Reply:
Moved JSON::JWK::Set to dependency

Reply:
This is inside rescure and if the JSON parse fails we want another error. This way we cant check if keys.blank? inside this function and need to return both